packages/patternfly-4/react-integration/demo-app-ts/src/components/demos/TableDemo/TableActionsDemo.tsx

```diff
@@ -0,0 +1,101 @@
+import React from 'react';
+import {
+  Table,
+  TableHeader,
+  TableBody,
+  sortable,
+  SortByDirection,
+  headerCol,
+  TableVariant,
+  expandable,
+  cellWidth,
+  IRow,
+  IActionsResolver,
+  IAction,
+  ISeparator
+} from '@patternfly/react-table';
+
+export class TableActionsDemo extends React.Component<{}, { columns: any, rows: IRow[] }> {
+  constructor(props) {
+    super(props);
+    this.state = {
+      columns: [
+        { title: 'Repositories', cellTransforms: [headerCol()] },
+        'Branches',
+        { title: 'Pull requests' },
+        'Workspaces',
+        'Last Commit'
+      ],
+      rows: [
+        {
+          cells: ['one', 'two', 'a', 'four', 'five'],
+          type: 'green'
+        },
+        {
+          cells: ['a', 'two', 'k', 'four', 'five']
+        },
+        {
+          cells: ['p', 'two', 'b', 'four', 'five'],
+          type: 'blue'
+        },
+        {
+          cells: ['5', '2', 'b', 'four', 'five']
+        }
+      ]
+    };
+  }
+
+  actionResolver(rowData, { rowIndex }) {
+    if (rowIndex === 1) {
+      return null;
+    }
+
+    const thirdAction =
+      rowData.type === 'blue'
+        ? [
+            {
+              isSeparator: true
+            },
+            {
+              title: 'Third action',
+              onClick: (event, rowId, rowData, extra) =>
+                console.log(`clicked on Third action, on row ${rowId} of type ${rowData.type}`)
+            }
+          ]
+        : [];
+
+    return [
+      {
+        title: 'Some action',
+        onClick: (event, rowId, rowData, extra) =>
+          console.log(`clicked on Some action, on row ${rowId} of type ${rowData.type}`)
+      },
+      {
+        title: <div>Another action</div>,
+        onClick: (event, rowId, rowData, extra) =>
+          console.log(`clicked on Another action, on row ${rowId} of type ${rowData.type}`)
+      },
+      ...thirdAction
+    ];
+  }
+
+  areActionsDisabled(rowData, { rowIndex }) {
+    return rowIndex === 3;
+  }
+
+  render() {
+    const { columns, rows } = this.state;
+    return (
+      <Table
+        caption="Actions Table"
+        cells={columns}
+        rows={rows}
+        actionResolver={() => null as (IAction | ISeparator)[]}
```

Comment:
Changed to this function declaration from `actionResolver={this.actionResolver}`. Was getting a type error here before... not sure if this will work

Reply:
yes... we can try to cast these callbacks from our interface correctly now.. so this becomes something like this:

```
import React from 'react';
import {
  Table,
  TableHeader,
  TableBody,
  sortable,
  SortByDirection,
  headerCol,
  TableVariant,
  expandable,
  cellWidth,
  IRow,
  IRowData,
  IExtra,
  IActionsResolver,
  IAction,
  ISeparator
} from '@patternfly/react-table';

export class TableActionsDemo extends React.Component<{}, { columns: any, rows: IRow[] }> {
  constructor(props) {
    super(props);
    this.state = {
      columns: [
        { title: 'Repositories', cellTransforms: [headerCol()] },
        'Branches',
        { title: 'Pull requests' },
        'Workspaces',
        'Last Commit'
      ],
      rows: [
        {
          cells: ['one', 'two', 'a', 'four', 'five'],
          type: 'green'
        },
        {
          cells: ['a', 'two', 'k', 'four', 'five']
        },
        {
          cells: ['p', 'two', 'b', 'four', 'five'],
          type: 'blue'
        },
        {
          cells: ['5', '2', 'b', 'four', 'five']
        }
      ]
    };
  }

  actionResolver(rowData: IRowData, { rowIndex }: IExtra) {
    if (rowIndex === 1) {
      return null;
    }

    const thirdAction =
      rowData.type === 'blue'
        ? [
            {
              isSeparator: true
            } as ISeparator,
            {
              title: 'Third action',
              // tslint:disable-next-line:no-shadowed-variable
              onClick: (event, rowId, rowData, extra) =>
                // tslint:disable-next-line:no-console
                console.log(`clicked on Third action, on row ${rowId} of type ${rowData.type}`)
            } as IAction
          ]
        : [];

    return [
      {
        title: 'Some action',
        // tslint:disable-next-line:no-shadowed-variable
        onClick: (event, rowId, rowData, extra) =>
          // tslint:disable-next-line:no-console
          console.log(`clicked on Some action, on row ${rowId} of type ${rowData.type}`)
      } as IAction,
      {
        title: <div>Another action</div>,
        // tslint:disable-next-line:no-shadowed-variable
        onClick: (event, rowId, rowData, extra) =>
          // tslint:disable-next-line:no-console
          console.log(`clicked on Another action, on row ${rowId} of type ${rowData.type}`)
      } as IAction,
      ...thirdAction
    ];
  }

  areActionsDisabled(rowData, { rowIndex }) {
    return rowIndex === 3;
  }

  render() {
    const { columns, rows } = this.state;
    return (
      <Table
        caption="Actions Table"
        cells={columns}
        rows={rows}
        actionResolver={this.actionResolver}
        areActionsDisabled={this.areActionsDisabled}
      >
        <TableHeader />
        <TableBody />
      </Table>
    );
  }
}
```

Reply:
Thank you Patrick!! This is very helpful, I can now see where and why you added those specific type assertions 🙂